src/snowflake/snowpark/async_job.py

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+from enum import Enum
+from typing import Dict, Iterator, List, Union
+
+import snowflake.snowpark
+from snowflake.connector import SnowflakeConnection
+from snowflake.connector.options import pandas
+from snowflake.snowpark._internal.utils import (
+    check_is_pandas_dataframe_in_to_pandas,
+    result_set_to_iter,
+    result_set_to_rows,
+)
+from snowflake.snowpark.row import Row
+
+
+class _AsyncDataType(Enum):
+    PANDAS = "pandas"
+    ROW = "row"
+    ITERATOR = "iterator"
+    PANDAS_BATCH = "pandas_batch"
+    COUNT = "count"
+    NONE_TYPE = "no_type"
+    UPDATE = "update"
+    DELETE = "delete"
+    MERGE = "merge"
+
+
+class AsyncJob:
```

Comment:
Let's add docstrings for this class, including the introduction, examples, caveats and notes. You can refer to other files for examples.